copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookIOUtils.java

```diff
@@ -61,8 +63,12 @@ public static ExternalRecord getExternalRecord(InputStream cblIs) throws RecordE
    */
   public static int getRecordLength(ExternalRecord externalRecord, int fileStructure) {
     int recordByteLength = 0;
+    Set<Integer> fieldPositions = new HashSet<>();
     for (ExternalField field : externalRecord.getRecordFields()) {
-      recordByteLength += field.getLen();
+      if (!fieldPositions.contains(field.getPos())) {
+        recordByteLength += field.getLen();
+        fieldPositions.add(field.getPos());
+      }
```

Comment:
Analysed the suggested fix but it calculates the record length incorrectly. And, it seems the record length in combination with the splits, gets the parsing of Cobol records done in a wrong manner and throws exceptions. 

Therefore, considering the use case suggested, where the existing length calculation fails, I have applied another approach to fix the issue.